cgk_calcium_tools/filehandler.py

```diff
@@ -118,28 +133,50 @@ def __init__(
             "duration": [],
             "frames_per_second": [],
         }
+
         loaded_meta_files = [] #this variable is used to don't load multiple times the same json
+        
         for mainf, subfolder, fpatter, outf in zip(
             lists_inputs["main_data_folder"],
             lists_inputs["data_subfolders"],
             lists_inputs["files_patterns"],
             lists_inputs["outputsfolders"],
         ):
-            if check_new_imputs:
-                # From raw data origin
-                files = glob(str(Path(mainf) / subfolder / fpatter), recursive=False)
-                assert len(files) > 0, "No file found for {}".format(
+            if check_new_inputs:
+                allFiles = glob(str(Path(mainf) / subfolder / fpatter),  recursive=True) #grabs all the files with fpatter.
+                files = [file for file in allFiles if skip_pattern not in Path(file).name] #filter skip_pattern files out
+                print(f'{len(files)} files found, {len(allFiles)-len(files)} file(s) skipped') #prints what it found
+                
+                #Error no file found. 
+                assert len(files) > 0, "No file(s) found for {}".format(
                     str(Path(mainf) / subfolder / fpatter)
                 )
                 metadata = {}
 
                 if single_file_match:
                     assert len(files) == 1, "Multiple files found for {}.".format(
                         str(Path(mainf) / subfolder / fpatter)
-                        # Include a better msg that explain what this error is and how to fix it. 
                     )
                 else:
                     files = [r for r in files if r not in meta["rec_paths"]]
+
+                # Initialize ipywidgets progress bar
+                progress_bar = IntProgress(
+                    value=0, 
+                    min=0,
+                    max=len(files), 
+                    style={
+                        'bar_color': '#3385ff',
+                        'description_width': 'auto'
+
+                    },
+                    description='Loading Files: 0/'+str(len(files)),
+                    layout=Layout(width='45%')
+                )
+            
+                # Display the progress bar with descriptions
+                display(progress_bar)                
```

Comment:
making a function for this code with the **max** as an input it will prevent duplicate code in the future